engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -535,8 +534,12 @@ public void runLocal(AppFactory appConfig) throws Exception
 
   public URLClassLoader loadDependencies()
   {
-    URLClassLoader cl = URLClassLoader.newInstance(launchDependencies.toArray(new URL[launchDependencies.size()]));
+    URL[] dependencies = launchDependencies.toArray(new URL[launchDependencies.size()]);
+
+    ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
The suggestion is to cache URL class loaders on the instance of the StramAppLauncher instance to avoid unnecessary instantiation of URL class loaders with the same dependencies. If dependencies change, the cache should be cleared.